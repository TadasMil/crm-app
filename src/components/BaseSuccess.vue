<script setup name="BaseSuccess" lang="ts">
import BaseIcon from './BaseIcon.vue'

defineProps<{
  title?: string
}>()
</script>

<template>
  <div
    class="flex h-full w-full flex-col items-center justify-center space-y-2"
  >
    <BaseIcon icon="success" class="h-16 w-16 text-success" />
    <p class="font-bold text-success">{{ title }}</p>
    <slot />
  </div>
</template>
