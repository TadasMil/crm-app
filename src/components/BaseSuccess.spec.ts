import { render } from '@testing-library/vue'
import { describe, expect, it, vi } from 'vitest'

import BaseSuccess from './BaseSuccess.vue'

describe('BaseSuccess.spec.js', () => {
  it('should render the component', () => {
    const { html } = render(BaseSuccess)

    expect(html()).toMatchSnapshot()
  })

  it('should render the component with custom title', () => {
    const { html } = render(BaseSuccess, {
      props: {
        title: 'Hello, please wait we are loading',
      },
    })

    expect(html()).toContain('Hello, please wait we are loading')
  })
})
