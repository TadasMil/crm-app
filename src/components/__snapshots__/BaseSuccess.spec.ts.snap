// Vitest Snapshot v1

exports[`BaseSuccess.spec.js > should render the component 1`] = `
"<div class=\\"flex h-full w-full flex-col items-center justify-center space-y-2\\"><svg class=\\"block flex-shrink-0 fill-current h-16 w-16 text-success\\">
    <use href=\\"/icons/dashboard.svg?#success\\" xlink:href=\\"/icons/dashboard.svg?#success\\"></use>
  </svg>
  <p class=\\"font-bold text-success\\"></p>
</div>"
`;
